pkg/ddl/ddl_running_jobs.go

```diff
@@ -53,40 +53,38 @@ func (j *runningJobs) clear() {
 	j.unfinishedSchema = make(map[string]map[string]struct{})
 }
 
-func (j *runningJobs) add(job *model.Job) {
+func (j *runningJobs) add(jobID int64, involvedSchemaInfos []model.InvolvingSchemaInfo) {
 	j.Lock()
 	defer j.Unlock()
-	j.processingIDs[job.ID] = struct{}{}
+	j.processingIDs[jobID] = struct{}{}
 	j.updateInternalRunningJobIDs()
 
-	if _, ok := j.unfinishedIDs[job.ID]; ok {
+	if _, ok := j.unfinishedIDs[jobID]; ok {
 		// Already exists, no need to add it again.
 		return
 	}
-	j.unfinishedIDs[job.ID] = struct{}{}
-	for _, info := range job.GetInvolvingSchemaInfo() {
+	j.unfinishedIDs[jobID] = struct{}{}
+	for _, info := range involvedSchemaInfos {
 		if _, ok := j.unfinishedSchema[info.Database]; !ok {
 			j.unfinishedSchema[info.Database] = make(map[string]struct{})
 		}
 		j.unfinishedSchema[info.Database][info.Table] = struct{}{}
 	}
 }
 
-func (j *runningJobs) remove(job *model.Job) {
+func (j *runningJobs) remove(jobID int64, involvedSchemaInfos []model.InvolvingSchemaInfo) {
 	j.Lock()
 	defer j.Unlock()
-	delete(j.processingIDs, job.ID)
+	delete(j.processingIDs, jobID)
 	j.updateInternalRunningJobIDs()
 
-	if job.IsFinished() || job.IsSynced() {
```

Comment:
reorg jobs will check whether there are jobs of same table-id are `in-processing` in `getReorgJob`, so for a table-id there shouldn't be 2 jobs running concurrently, so it's ok to delete those unfinishedXXX all the time

Reply:
@D3Hunter Is it not necessary to distinguish `processingIDs` from `unfinishedIDs`?

Reply:
like what we have discussed, `runningJobs.remove` will only called when the job is in final state like synced/paused/cancelled or it's done by another owner, before that another job operating the same table-id or name should be blocked